core/chains/evm/chain.go

```diff
@@ -148,20 +152,19 @@ func newChain(dbchain types.Chain, opts ChainSetOpts) (*chain, error) {
 
 	headBroadcaster.Subscribe(logBroadcaster)
 
-	c := chain{
-		utils.StartStopOnce{},
-		chainID,
-		cfg,
-		client,
-		txm,
-		l,
-		headBroadcaster,
-		headTracker,
-		logBroadcaster,
-		balanceMonitor,
-		opts.KeyStore,
-	}
-	return &c, nil
+	return &chain{
```

Comment:
I think that actually sets off the go vet linter no? composite literals unkeyed or something

Reply:
Yeah, it can be error prone since they may silently change out from under you.

Reply:
> because the code won't compile

This assumption doen't hold and is why it 'may silently change out from under you'. For example, reordering of fields, particularly with _untyped_ constants.

> otherwise you'll just unknowingly pass zero values

This is less risky, both because zero values being usable by default is idiomatic in go, but more importantly because when fields are added to a struct it is expected that users may have to react, whereas it is unintuitive and surprising to depend on the internal struct field ordering at all in the first place.

There is more nuance to consider, like if an exported API boundary is being crossed, but in cases like this with internal constructors I generally favor naming to take _advantage_ of zero values being meaningful, especially to avoid passing things like `sync.RWMutex{}` and `utils.StartStopOnce{}`.